crates/topos/src/components/tce/services/push_certificate.rs

```diff
@@ -31,143 +31,27 @@ pub(crate) async fn check_delivery(
-                            target_subnet_ids: vec![[2u8; SUBNET_ID_LENGTH].into()],
-                            positions: vec![]
-                        }),
-                        source_checkpoint: None
-                    }.into()
-                };
-
-                let response = client.watch_certificates(in_stream).await.map_err(|_| {
-                    (
-                        peer_string.clone(),
-                        "Unable to execute the watch_certificates on TCE api",
-                    )
-                })?;
-                let mut resp_stream = response.into_inner();
-                async move {
-                    while let Some(received) = resp_stream.next().await {
-                        let received = received.unwrap();
-                        if let Some(Event::CertificatePushed(CertificatePushed {
-                            certificate: Some(certificate),
-                            ..
-                        })) = received.event
-                        {
-                            // unwrap is safe because we are sure that the certificate is present
-                            if certificate_id == certificate.id.unwrap() {
-                                debug!("Received the certificate on {}", peer_string);
-                                return Ok(());
-                            }
-                        }
-                    }
-
-                    Err((peer_string.clone(), "didn't receive any certificate"))
-                }
-                .await
-            }));
-        }
-
-        let mut client = ApiServiceClient::connect(random_peer.clone())
-            .await
-            .map_err(|_| vec![format!("Unable to connect to the TCE api on {random_peer}")])?;
-
-        // submit a certificate to one node
-        _ = client
-            .submit_certificate(SubmitCertificateRequest {
-                certificate: Some(pushed_certificate.into()),
-            })
-            .await
-            .map_err(|_| {
-                vec![format!(
-                    "Unable to submit the certificate to the TCE api on {random_peer}"
-                )]
-            })?;
-
-        tokio::time::sleep(Duration::from_secs(timeout_broadcast)).await;
-        let mut errors = vec![];
-
-        join_all(join_handlers)
-            .await
-            .iter()
-            .for_each(|result| match result {
-                Err(_) => {
-                    errors.push("Unable to properly execute command".to_string());
-                }
-                Ok(Err((peer, error))) => {
-                    errors.push(format!("{peer} {error}"));
-                }
-                _ => {}
-            });
-
-        if errors.is_empty() {
-            Ok(())
-        } else {
-            Err(errors)
-        }
-    })
-    .await
-    .map_err(|error| {
-        info!("Timeout reached: {:?}", error);
-        error
-    })
+) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Is this the right error type? Consider adding new error variants as needed to reduce the `map_err` calls a bit.

Reply:
I am reluctant to create a new error type for this - it is test code after all

Reply:
As I returned function to tce services, I'll create new error type for it

Reply:
Introducing custom error type has to be done on a tce::commands module level. It is a topic for itself and another PR